s2fft/utils/signal_generator.py

```diff
@@ -94,29 +99,39 @@ def generate_flm(
 
         using_torch (bool, optional): Desired frontend functionality. Defaults to False.
 
+        size (tuple[int, ...] | int | None, optional): Shape of realisations.
+
     Returns:
         np.ndarray: Random set of spherical harmonic coefficients.
 
     """
-    flm = np.zeros(samples.flm_shape(L), dtype=np.complex128)
+    # always turn size into a tuple of int
+    if size is None:
+        size = ()
+    elif isinstance(size, int):
+        size = (size,)
+    elif not (isinstance(size, tuple) and all(isinstance(_, int) for _ in size)):
```

Comment:
```suggestion
    elif not (isinstance(size, tuple) and all(isinstance(i, int) for i in size)):
```
Nitpick, but I think conventionally `_` is used to indicate unused variables in assignments / loops, whereas here the loop variable is being used.